libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/STM32Fxx/NetworkInterface.c

```diff
@@ -1288,27 +1145,41 @@ BaseType_t xReturn;
 }
 /*-----------------------------------------------------------*/
 
+/* Uncomment this in case BufferAllocation_1.c is used. */
+
+#define niBUFFER_1_PACKET_SIZE		1536
```

Comment:
I think this is the default max frame size, per RFC, correct? Seems like it should be a global define somewhere instead.

Reply:
Hello @dcgaws : thanks for these useful comments.

Two typos "command" and "of PHY" : corrected

> xRemainingTime = ( TickType_t ) pdMS_TO_TICKS( 1000UL );
> Should this value be pulled out to a configuration header?

Yes you are right.  Replaced with three macro's:

    #define phyPHY_MAX_RESET_TIME_MS            1000uL

    #define phyPHY_MAX_NEGOTIATE_TIME_MS        3000uL

    #define phySHORT_DELAY_MS                   50uL

> const TickType_t xDescriptorWaitTime = pdMS_TO_TICKS( 250 );
> Should this value be configurable?

Yes, also:

    #ifned niDESCRIPTOR_WAIT_TIME_MS
        #define niDESCRIPTOR_WAIT_TIME_MS       250uL
    #endif


>     /* The expected ID for the 'LAN8742A'  is 0x0007c130. */
> 
> Are there references to the standard?

What do you mean with this question?

... I added all 100 Mbps PHY's that I encountered on various evaluation boards.
Their register layout and the register bits vary from chip to chip. The 28-bit ID is used to select the right layout.

> BaseType_t xPhyFixedValue( EthernetPhy_t *pxPhyObject, uint32_t ulPhyMask )
> 
> Could function comment headers please be added? This one in particular made
> me think of it, because I don't understand what it does based on the name.

You are right, the function name is very cryptic. I added several comments:

~~~c
/* xPhyFixedValue(): this function is called in case auto-negotiation is disabled.
The caller has set the values in 'xPhyPreferences' (ucDuplex and ucSpeed).
The PHY register phyREG_00_BMCR will be set for every connected PHY that matches
with ulPhyMask. */

/* xPhyStartAutoNegotiation() is the alternative xPhyFixedValue():
It sets the BMCR_AN_RESTART bit and waits for the auto-negotiation completion
( phyBMSR_AN_COMPLETE ). */
~~~

> How do you recommend that we test this?

It is a pity that the AWS/FreeRTOS repo doesn't have a demo project that uses the STM32Fxx. In the old FreeRTOS demo's there are STM32Fx projects. I could adapt one such project to include the new driver.

I tested it with an Eclipse/Makefile project. But that is too dirty to be made public ( full of temporary ad-hoc tests ).

> configASSERT( xTXDescriptorSemaphore );
> 
> Should this be a runtime check rather than an assert? I'm wondering what
> will happen here in production code, following resource exhaustion, with
> assert compiled out.

Absolutely right. I added the variable `xMacInitStatus`, also seen in other network drivers :

~~~c
    eMACInit,   /* Must initialise MAC. */
    eMACPass,   /* Initialisation was successful. */
    eMACFailed, /* Initialisation failed. */
~~~

> #define niBUFFER_1_PACKET_SIZE        1536
> 
> I think this is the default max frame size, per RFC, correct?

Yes it is some boundary. I had to look it up [here](https://en.wikipedia.org/wiki/Ethernet_frame).

A buffer of 1536 bytes is larger than necessary, all +TCP projects use an MTU of at most 1500 bytes. So no more than 1514 bytes are used.
The good thing about 1536 is its **super alignment**: it is a multiple of 512 ( 9 bits ). Alignment is important for both DMA and memory caching. Two network buffers may never share the same caching row.

> #define niBUFFER_1_PACKET_SIZE        1536
> Seems like it should be a global define somewhere instead.

The header file `stm32fxx_hal_eth.h` already defines `ETH_MAX_PACKET_SIZE` with the same value. Let's use that macro here as well.

Reply:
In a meanwhile, I updated the PR in a new commit:

"FreeRTOS+TCP : STM32Fxx network interface Updated, v2 "